File: Android/app/src/main/java/com/edu/primary/constants/AppConstants.java
package com.edu.primary.constants;

public class AppConstants {
    // 数据库相关
    public static final String DATABASE_NAME = "primary_education_database";
    
    // SharedPreferences相关
    public static final String PREFS_USER = "user_prefs";
    public static final String PREFS_AI = "ai_prefs";
    public static final String KEY_USER_ID = "user_id";
    public static final String KEY_USERNAME = "username";
    public static final String KEY_API_KEY = "deepseek_api_key";
    
    // AI相关
    public static final int MAX_HISTORY_MESSAGES = 50;
    public static final String DEEPSEEK_BASE_URL = "https://api.deepseek.com/";
    public static final String DEEPSEEK_MODEL = "deepseek-chat";
    public static final double DEEPSEEK_TEMPERATURE = 0.7;
    public static final int DEEPSEEK_MAX_TOKENS = 2000;
    public static final String DEEPSEEK_API_ENDPOINT = "v1/chat/completions";
    public static final String HEADER_AUTHORIZATION = "Authorization";
    public static final String HEADER_CONTENT_TYPE = "Content-Type";
    public static final String BEARER_PREFIX = "Bearer ";
    public static final String CONTENT_TYPE_JSON = "application/json";
    
    // 科目ID
    public static final int SUBJECT_CHINESE = 1;
    public static final int SUBJECT_MATH = 2;
    public static final int SUBJECT_ENGLISH = 3;
    
    // 年级范围
    public static final int MIN_GRADE = 1;
    public static final int MAX_GRADE = 6;
    
    // 题目类型
    public static final String QUESTION_TYPE_SINGLE_CHOICE = "single_choice";
    public static final String QUESTION_TYPE_MULTIPLE_CHOICE = "multiple_choice";
    public static final String QUESTION_TYPE_FILL_BLANK = "fill_blank";
    public static final String QUESTION_TYPE_JUDGMENT = "judgment";
    
    // 难度范围
    public static final int MIN_DIFFICULTY = 1;
    public static final int MAX_DIFFICULTY = 5;
    
    // 练习题目数量（根据BuildConfig动态设置）
    public static final int DEBUG_QUESTIONS_PER_PRACTICE = 5;
    public static final int RELEASE_QUESTIONS_PER_PRACTICE = 40;
    
    // 消息角色
    public static final String ROLE_USER = "user";
    public static final String ROLE_ASSISTANT = "assistant";
    public static final String ROLE_SYSTEM = "system";
    
    // 错误消息键
    public static final String ERROR_USERNAME_EXISTS = "username_exists";
    public static final String ERROR_USER_NOT_FOUND = "user_not_found";
    public static final String ERROR_PASSWORD_WRONG = "password_wrong";
    public static final String ERROR_API_KEY_NOT_SET = "api_key_not_set";
    public static final String ERROR_LOAD_FAILED = "load_failed";
    public static final String ERROR_NETWORK = "network_error";
    
    private AppConstants() {
        // Prevent instantiation
    }
}

File: Android-1/app/src/main/java/com/edu/primary/util/AppConstants.kt
package com.edu.primary.util

/**
 * 应用全局常量：SharedPreferences 键名、学科/年级/题型/难度范围、练习题目数量、角色与错误码等。
 */
object AppConstants {
    const val PREFS_USER = "user_prefs"
    const val PREFS_AI = "ai_prefs"
    const val KEY_USER_ID = "user_id"
    const val KEY_USERNAME = "username"
    const val KEY_API_KEY = "deepseek_api_key"
    const val MAX_HISTORY_MESSAGES = 50
    const val SUBJECT_CHINESE = 1
    const val SUBJECT_MATH = 2
    const val SUBJECT_ENGLISH = 3
    const val MIN_GRADE = 1
    const val MAX_GRADE = 6
    const val QUESTION_TYPE_SINGLE_CHOICE = "single_choice"
    const val QUESTION_TYPE_MULTIPLE_CHOICE = "multiple_choice"
    const val QUESTION_TYPE_FILL_BLANK = "fill_blank"
    const val QUESTION_TYPE_JUDGMENT = "judgment"
    const val MIN_DIFFICULTY = 1
    const val MAX_DIFFICULTY = 5
    const val DEBUG_QUESTIONS_PER_PRACTICE = 5
    const val RELEASE_QUESTIONS_PER_PRACTICE = 40
    const val ROLE_USER = "user"
    const val ROLE_ASSISTANT = "assistant"
    const val ROLE_SYSTEM = "system"
    const val ERROR_USERNAME_EXISTS = "username_exists"
    const val ERROR_USER_NOT_FOUND = "user_not_found"
    const val ERROR_PASSWORD_WRONG = "password_wrong"
    const val ERROR_API_KEY_NOT_SET = "api_key_not_set"
}
